### PR TITLE
Make workflow objects read only in frontend

### PR DESCRIPTION
**Pull request by thomtrp**
Expected behavior:
- workflows can be added and deleted. Only name field is editable
- versions and runs cannot be added nor deleted. No fields are editable

Added two new utils for those needs:
- `isReadOnlyObject` the similar logic between remote objects, versions and runs
- `isFieldReadonlyFromObjectMetadataName` to easily block field edition from object context

### CODE REVIEW

**Comment by charlesBochet** [comment]
@thomtrp the CI is red because of the linter!

**Comment by thomtrp** [comment]
> the CI is red because of the linter!

@charlesBochet CI is fixed

**Comment by charlesBochet** [comment]
> > the CI is red because of the linter!
> 
> @charlesBochet CI is fixed

You bet :D

**Comment by thomtrp** [comment]
> > > the CI is red because of the linter!
> > 
> > 
> > @charlesBochet CI is fixed
> 
> You bet :D

For good this time 😂

File: packages/twenty-front/src/modules/object-metadata/utils/isReadOnlyObject.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+import { isWorkflowSubObject } from '@/object-metadata/utils/isWorkflowSubObject';
+
+export const isReadOnlyObject = (
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/utils/isReadOnlyObject.ts:4]: isReadOnlyObjectMetadata

File: packages/twenty-front/src/modules/object-metadata/utils/isWorkflowSubObject.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+
+export const isWorkflowSubObject = (objectMetadataNameSingular?: string) =>
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/utils/isWorkflowSubObject.ts:3]: ObjectMetadata

File: packages/twenty-front/src/modules/object-record/record-field/hooks/useIsFieldReadOnly.ts [hunk 3]
```diff
@@ -2,15 +2,22 @@ import { useContext } from 'react';
...
 
 export const useIsFieldReadOnly = () => {
   const { fieldDefinition } = useContext(FieldContext);
 
+  const { metadata } = fieldDefinition;
+
   return (
-    fieldDefinition.metadata.fieldName === 'noteTargets' ||
-    fieldDefinition.metadata.fieldName === 'taskTargets' ||
+    metadata.fieldName === 'noteTargets' ||
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/hooks/useIsFieldReadOnly.ts:14]: move it to your new util

File: packages/twenty-front/src/modules/object-record/record-field/utils/isFieldReadonlyFromObjectMetadataName.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { isWorkflowSubObject } from '@/object-metadata/utils/isWorkflowSubObject';
+
+export const isFieldReadonlyFromObjectMetadataName = (
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/utils/isFieldReadonlyFromObjectMetadataName.ts:4]: I would put these hooks into object-metadata?

**Review by charlesBochet — COMMENTED** [review]
LGTM!

**Review by charlesBochet — APPROVED** [review]

File: packages/twenty-front/src/modules/object-record/record-field/utils/isFieldReadonlyFromObjectMetadataName.ts (hunk 4 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/object-record/record-field/utils/isFieldReadonlyFromObjectMetadataName.ts:4]: checked with you. Leaving them here

**Pull request merged**